src/detect.ts

```diff
@@ -14,3 +15,23 @@ export async function detectPackageManager(cwd = process.cwd()) {
   const agent = (result ? LOCKS[path.basename(result)] : null)
   return agent
 }
+
+/**
+ * Detect yarn berry version (2+) based on the .yarn folder and the
+ * package.json file.
+ * Also checks in parent directories for yarn workspaces.
+ */
+export async function isYarnBerry(cwd = process.cwd()) {
+  const result = await findUp('.yarn', { cwd, type: 'directory' })
+  if (!result) return false
+
+  // In yarn v1 was also the possibility that a .yarn folder exists.
+  // Check package.json for yarn version.
+  const pkg = await readPackage({ cwd: path.dirname(result) })
```

Comment:
I don't want to add another dependency. We could probably improve `detectPackageManager` instead, as not only for yarn, other package managers may also have breakages across versions (e.g. `pnpm@6` vs `pnpm@7`)

You can reference to:
https://github.com/antfu/ni/blob/f814583e96ac55a9ad647b8ff8b31be62fa27e5d/src/detect.ts#L17-L49